File: Programmers/Lv.1/문자열 나누기/140108.java
class Solution {
    public int solution(String s) {
        int answer = 1;
        
        int xCount = 0;
        int noxCount = 0;
        char str = s.charAt(0);
        for (int i = 0; i < s.length() - 1; i++) {
            if (s.charAt(i) == str) {
                xCount++;
            } else {
                noxCount++;
            }
            
            if (xCount == noxCount) {
                answer++;
                xCount = 0;
                noxCount = 0;
                str = s.charAt(i + 1);
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/문자열 나누기/140108.js
function solution(s) {
    let answer = 1;
    
    let xCount = 0, noxCount = 0;
    let str = s.charAt(0);
    for (let i = 0; i < s.length - 1; i++) {
        if (s.charAt(i) === str) {
            xCount++;
        } else {
            noxCount++;
        }
        
        if (xCount === noxCount) {
            answer++;
            xCount = 0, noxCount = 0;
            str = s.charAt(i + 1);
        }
    }
    
    return answer;
}
